File: src/main/java/implement/general/AbstractInformable.java
package implement.general;

import framework.general.Informable;

/**
 * Removes some boilerplate code from other classes.
 */
public abstract class AbstractInformable implements Informable {

    protected String name, description;

    public AbstractInformable(String name, String description) {
        this.name = name;
        this.description = description;
    }

    @Override
    public String getDescription() {
        return description;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: src/main/java/implement/general/AbstractInformable.scala
package implement.general

import framework.general.Informable

/**
 * Removes some boilerplate code from other classes.
 */
abstract class AbstractInformable(private var name: String, private var description: String) extends Informable {

  def this() = this(null, "")

  override def getDescription: String = description
  override def getName: String = if(name == null) getClass.getName else name
}